test/Jsr292/src/com/ibm/j9/jsr292/MethodHandleTest.java

```diff
@@ -565,75 +491,70 @@ public void test_asSpreader_CrossPackage_Virtual() throws Throwable{
 		MethodHandle mh = MethodHandles.lookup().findVirtual(PackageExamples.class,"addPublic",MethodType.methodType(int.class,int.class,int.class));
 		PackageExamples g = new PackageExamples();
 		mh = mh.bindTo(g);
-		boolean wrongMethodTypeExceptionThrown = false; 
-		
-		try {
-			mh = mh.asSpreader(String[].class, 2);
-		}
-		catch(WrongMethodTypeException e) {
-			wrongMethodTypeExceptionThrown = true;
-		}
-		
-		AssertJUnit.assertTrue(wrongMethodTypeExceptionThrown);
+		mh = mh.asSpreader(String[].class, 2);
+		Assert.fail("The test case failed to throw an WrongMethodTypeException in the case of the wrong method type");
 	}
 	
 	/**
 	 * Negative test : asSpreader test using a cross-package static method and wrong method type
 	 * @throws Throwable
 	 */
-	@Test(groups = { "level.extended" })
+	@Test(expectedExceptions = IllegalArgumentException.class, groups = { "level.extended" })
 	public void test_asSpreader_CrossPackage_Static_WrongType() throws Throwable{
 		MethodHandle mh = MethodHandles.lookup().findStatic(PackageExamples.class,"addPublicStatic",MethodType.methodType(int.class,int.class,int.class));
 		mh = mh.asSpreader(int[].class, 2);
-		boolean illegalArgumentExceptionThrown = false; 
-		
-		try {
-			mh = mh.asSpreader(String[].class, 2);
-		}
-		catch(IllegalArgumentException e) {
-			illegalArgumentExceptionThrown = true;
-		}
-		
-		AssertJUnit.assertTrue(illegalArgumentExceptionThrown);
+		mh = mh.asSpreader(String[].class, 2);
+		Assert.fail("The test case failed to throw an IllegalArgumentException in the case of the wrong method type");
 	}
 	
 	/**
 	 * Negative test : asSpreader test using cross package static method and a wrong argument count
 	 * @throws Throwable
 	 */
-	@Test(groups = { "level.extended" })
+	@Test(expectedExceptions = IllegalArgumentException.class, groups = { "level.extended" })
 	public void test_asSpreader_CrossPackage_Static_WrongArgCount() throws Throwable{
 		MethodHandle mh = MethodHandles.lookup().findStatic(PackageExamples.class,"addPublicStatic",MethodType.methodType(int.class,int.class,int.class));
-		
-		boolean illegalArgumentException = false; 
-		
-		try {
-			mh = mh.asSpreader(int[].class, 3);
-		}
-		catch(IllegalArgumentException e) {
-			illegalArgumentException = true;
-		}
-		
-		AssertJUnit.assertTrue(illegalArgumentException);
+		mh = mh.asSpreader(int[].class, 3);
+		Assert.fail("The test case failed to throw an IllegalArgumentException in the case of the wrong argument count");
 	}
 	
 	/******************************
 	 * Tests for MethodHandle arity
 	 * ****************************/
 	/**
+	 * Create MethodHandle with highest possible arity and use findConstructor (253)
+	 * @throws Throwable
+	 */
+	@Test(groups = { "level.extended" })
+	public void test_findConstructor_ArityLimit() throws Throwable {
+		MethodHandle mh = MethodHandles.lookup().findConstructor(HelperConstructorClass.class, helperMethodType_void_253int);
+		mh = mh.asSpreader(int[].class, 253);
```

Comment:
Does this need an assert?  Or is not throwing an exception sufficient?

Reply:
This is a normal test case that should pass in any case; otherwise, any unexpected exception will be captured there. 

In addition, we've got test_findConstructor_ArityLimit_TooHigh() to capture the exception in the corner case.